packages/flame/lib/src/game/base_game.dart

```diff
@@ -156,38 +147,8 @@ class BaseGame extends Game with FPSCounter {
   /// This methods is an async operation since it await the `onLoad` method of the component. Nevertheless, this method only need to be waited to finish if by some reason, your logic needs to be sure that the component has finished loading, otherwise, this method can be called without waiting for it to finish as the BaseGame already handle the loading of the component.
   ///
   /// *Note:* Do not add components on the game constructor. This method can only be called after the game already has its layout set, this can be verified by the [hasLayout] property, to add components upon a game initialization, the [onLoad] method can be used instead.
-  Future<void> add(Component c) async {
-    assert(
-      hasLayout,
-      '"add" called before the game is ready. Did you try to access it on the Game constructor? Use the "onLoad" method instead.',
-    );
-    prepare(c);
-    final loadFuture = c.onLoad();
-
-    if (loadFuture != null) {
-      await loadFuture;
-    }
-    _addLater.add(c);
-  }
-
-  /// Prepares and registers a list of components to be added on the next game tick
-  void addAll(Iterable<Component> components) {
-    components.forEach(add);
-  }
-
-  /// Marks a component to be removed from the components list on the next game tick
-  void remove(Component c) {
-    _removeLater.add(c);
-  }
-
-  /// Marks a list of components to be removed from the components list on the next game tick
-  void removeAll(Iterable<Component> components) {
-    _removeLater.addAll(components);
-  }
-
-  /// Marks all existing components to be removed from the components list on the next game tick
-  void clear() {
-    _removeLater.addAll(components);
+  Future<void> add(Component c) {
+    return components.addChild(c);
```

Comment:
I think that if we have `add` here, we should also have the other methods proxied.

Reply:
I thought about it, but the problem is there are so many collection methods, so where do we draw the line?

Reply:
I would say that the ones we had was a pretty good place to draw the line, but I do see your point.

Reply:
Now when I have thought about it a bit more I think I actually prefer the proxies removed, like you have done (possibly adding `addAll` too like you said). Since removal can just as well be done on a component level for single components, and the rest might be clearer if they are on the collection. I think `add` is good to keep though since it is so heavily used.

Reply:
add, addAll, remove, clear? I thought it could be confusing, specially game.clear vs game.components.clear. but no strong opinion, I can definitely add them again as I said. also the baseComponent didn't expose all those, so we should unify that as well. but let's here what everyone has to say

Reply:
I will add addAll back, and only that. we can reconsider the api later of course